app/jobs/geocode_firm_job.rb

```diff
@@ -0,0 +1,19 @@
+class GeocodeFirmJob < ActiveJob::Base
+  def perform(firm)
+    results = Geocoder.search(firm.full_street_address)
+
+    if results.any?
+      Stats.increment('radsignup.geocode.firm.success')
+
+      firm.latitude = results.first.latitude
```

Comment:
As discussed in person - this is a great candidate for pushing into the domain. `Firm#geocode!` or some such. I suggested a bang method as it mutates your `Firm` instance. But I'm easy on that.